### PR TITLE
Make all interactive

### PR DESCRIPTION
**Pull request by guysalt**

### CODE REVIEW

File: install_mac.py [hunk 1]
```diff
@@ -103,11 +103,16 @@ def install_brew_packages(disable: List[str]):
...
+        if p in disable:
+            logger.info(f'skipping {p}')
+            continue
+
+        if p not in brew_list:
+            confirm_install(f'install {p}', brew['install', p, '--force'])
 
-    git('lfs', 'install')
-    sudo('git', 'lfs', 'install', '--system')
+    if 'git-lfs' not in brew_list and 'git-lfs' in brew('list').split('\n'):
```
Comment (doronz88) [inline install_mac.py:113]: use the `brew_list`

File: install_mac.py [hunk 2]
```diff
@@ -103,11 +103,16 @@ def install_brew_packages(disable: List[str]):
...
+        git('lfs', 'install')
```
Comment (doronz88) [inline install_mac.py:114]: missed a spot? or do we assume git lfs is a mandatory requirement?

File: install_mac.py [hunk 3]
```diff
@@ -145,73 +152,96 @@ def install_brew_packages(disable: List[str]):
...
             brew('install', '--cask', cask, '--force')
 
-    logger.info('installing blacktop\'s ipsw')
-    brew('install', 'blacktop/tap/ipsw')
+    confirm_install('install ipsw', brew['install', 'blacktop/tap/ipsw'])
 
-    logger.info('setting default browser to google chrome')
-    local['defaultbrowser']('chrome')
+    if 'google-chrome' not in brew_list and 'defaultbrowser' in brew_list:
+        brew_list = brew('list').split('\n')
```
Comment (doronz88) [inline install_mac.py:158]: use the already existing `brew_list`

File: install_mac.py [hunk 4]
```diff
@@ -145,73 +152,96 @@ def install_brew_packages(disable: List[str]):
...
         global AUTOMATION_MODE
         AUTOMATION_MODE = True
+
+
+class BaseCommand(click.Command):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params[:0] = [
+            click.Option(('-a', '--automated'), is_flag=True, callback=set_automation, expose_value=False,
+                         help='do everything without prompting')]
```
Comment (doronz88) [inline install_mac.py]: add an `(unless certain remvals are required)`

File: install_mac.py [hunk 5]
```diff
@@ -145,73 +152,96 @@ def install_brew_packages(disable: List[str]):
...
+
+
+@click.group(help='use -a/--automated flag for doing every command without prompts')
```
Comment (doronz88) [inline install_mac.py]: - use docstring for help message instead
- this should doc the command and not a specific option. for that we'll have the specific option's own help

**Review by doronz88 — CHANGES_REQUESTED** [review]

File: install_mac.py (hunk 1 above)
Comment (guysalt) [reply install_mac.py:113]: i want to check if `git-lfs` installed just now and not been install before...

File: install_mac.py (hunk 2 above)
Comment (guysalt) [reply install_mac.py:114]: check if `git-lfs` just installed through brew

File: install_mac.py (hunk 3 above)
Comment (guysalt) [reply install_mac.py:158]: this line comes after the prompting section so need to do brew list to check what installed now

File: install_mac.py (hunk 5 above)
Comment (guysalt) [reply install_mac.py]: I wanted to do that if anyone call the group (`python3 install_mac.py`) without call a specific subcommand he will see the `automated` option also. And i did not find out how to do so :(

Comment (guysalt) [reply install_mac.py]: anyway, i remove it now...

File: install_mac.py (hunk 4 above)
Comment (guysalt) [reply install_mac.py]: done

**Review by doronz88 — APPROVED** [review]

**Pull request merged**